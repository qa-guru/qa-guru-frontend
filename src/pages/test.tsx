import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
