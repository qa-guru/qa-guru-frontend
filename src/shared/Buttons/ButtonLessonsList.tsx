import React from "react";
import { Button, Typography } from "@mui/material";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import { useNavigate, useParams } from "react-router-dom";
import { primary } from "../../theme/colors";

const style = {
  button: { mb: "25px", color: primary.main },
  icon: { mr: "10px" },
};

const ButtonLessonsList: React.FC = () => {
  const { trainingId } = useParams();
  const navigate = useNavigate();

  return (
    <Button
      variant="outlined"
      sx={style.button}
      onClick={() => navigate(`/training/${trainingId}`)}
    >
      <ArrowBackIcon sx={style.icon} />
      <Typography textTransform="none" variant="subtitle1">
        К списку уроков
      </Typography>
    </Button>
  );
};

export default ButtonLessonsList;
