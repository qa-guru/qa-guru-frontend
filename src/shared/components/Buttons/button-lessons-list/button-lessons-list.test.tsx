import { MemoryRouter } from "react-router-dom";
import ButtonLessonsList from "./button-lessons-list";
import { render } from "../../../../test/utils-test";

describe("ButtonLessonsList", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ButtonLessonsList />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
