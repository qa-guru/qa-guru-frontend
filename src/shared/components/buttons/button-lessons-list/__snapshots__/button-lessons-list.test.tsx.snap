// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonLessonsList > renders correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-86zk83-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-qn5tej-MuiSvgIcon-root"
      data-testid="ChevronLeftIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body2 css-xw5kh4-MuiTypography-root"
    >
      К списку уроков
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
